lib/sqlalchemy/sql/elements.py

```diff
@@ -2527,7 +2575,7 @@ class Null(SingletonConstant, roles.ConstExprRole[None], ColumnElement[None]):
     _singleton: Null
 
     @util.memoized_property
-    def type(self):
+    def type(self) -> NullType:
```

Comment:
Mypy says it is incompatible with supertype `ColumnElement`, which (provided `TYPE_CHECKING == True`) has an attribute `type: TypeEngine[_T]` defined. Putting the same return type doesn't work. Possibly a decorator problem? Same goes for other `type()` overrides returning `Boolean` and `TypeEngine[_T]`